browse-next/src/components/DeviceSetupDefineMask.vue

```diff
@@ -0,0 +1,823 @@
+    computeImageDimensions();
+    generateMask();
+  });
+});
+
+const loadLatestStatusFrame = async () => {
+  if (device.value && device.value.type === "thermal") {
+    const latestStatus = await getLatestStatusRecordingForDevice(
+      device.value.id,
+      device.value.groupId
+    );
+    if (latestStatus) {
+      latestStatusRecording.value = latestStatus;
+    }
+  }
+};
+
+onBeforeUnmount(() => {
+  window.removeEventListener("resize", computeImageDimensions);
+  window.removeEventListener("resize", generateMask);
+  window.removeEventListener("resize", clearMask);
+});
+
+function normalise(x: number, y: number): Point {
+  const normalisedX = x / cptvFrameWidth.value;
+  const normalisedY = y / cptvFrameHeight.value;
+
+  return {
+    x: normalisedX,
+    y: normalisedY,
+  };
+}
+
+function pointSelect(event: MouseEvent): void {
+  if (creatingRegion.value && !enableAddRegionsButton.value) {
+    const { left, top } = container.value.getBoundingClientRect();
+    const clickedX = event.clientX - left;
+    const clickedY = event.clientY - top;
+    const values = normalise(clickedX, clickedY);
+    points.value.push({ x: values.x, y: values.y });
+    drawPolygon();
+  }
+}
+
+function drawRegionIndices() {
+  const canvasElement = canvas.value;
+  const context = canvasElement.getContext("2d");
+  const fontSize = 18 * devicePixelRatio.pixelRatio.value;
+  context.font = `bold ${fontSize}px Arial`;
+  for (let i = 0; i < regionsArray.value.length; i++) {
+    const regionData = regionsArray.value[i].regionData;
+    const firstPoint = regionData[0];
+    const text = `${i + 1}`;
+    const textMetrics = context.measureText(text);
+    const textWidth = textMetrics.width;
+    const circleRadius = devicePixelRatio.pixelRatio.value * 14; // Radius based on half of the text width
+
+    const circleX = firstPoint.x * canvas.value.width;
+    const circleY =
+      firstPoint.y * canvas.value.height +
+      2 * devicePixelRatio.pixelRatio.value;
+
+    context.fillStyle = "rgba(13, 110, 253)";
+    context.beginPath();
+    context.arc(circleX, circleY, circleRadius, 0, Math.PI * 2);
+    context.closePath();
+    context.fill();
+
+    context.fillStyle = "#ffffff";
+    context.fillText(
+      text,
+      circleX - textMetrics.width / 2,
+      circleY + fontSize / 2
+    );
+  }
+}
+
+function drawPolygon(): void {
+  const canvasElement = canvas.value;
+  const context = canvasElement.getContext("2d");
+  context.lineWidth = 4 * devicePixelRatio.pixelRatio.value;
+  context.strokeStyle = "rgba(13, 110, 253, 1)";
+  context.beginPath();
+  for (let i = 0; i < points.value.length; i++) {
+    const x = points.value[i].x * canvas.value.width;
+    const y = points.value[i].y * canvas.value.height;
+    if (i === 0) {
+      context.moveTo(x, y);
+    } else {
+      context.lineTo(x, y);
+    }
+  }
+  context.stroke();
+
+  context.fillStyle = "rgba(13, 110, 253, 1)";
+  for (let i = 0; i < points.value.length; i++) {
```

Comment:
In this kind of instance where you don't actually need the index of the point `for (const point of points.value) { ... }` is a more readable form.

Reply:
Does this instance not depend on the index? I'm not sure how you'd achieve the same behaviour without including the index